### PR TITLE
Jeff dev

### PR DESCRIPTION
**Pull request by jeffkinnison**
Four things here:

1. The call to optimizer.optimizer_from_flags() was being passed a learning rate, but it takes no arguments.
2. `train.py` wouldn't work on my machine without a shebang invoking the correct Python interpreter. This change should be transparent to everyone else.
3. `setup.py` wasn't installing all of the subpackages. Now it does :P
4. Requirements. So many requirements.

### CODE REVIEW

**Comment by jeffkinnison** [comment]
Additionally,

5. Merge changes committed by Hanyu
6. Update `_mpi` scripts to be `_parallel`
